src/lib/storage/table.hpp

```diff
@@ -182,8 +185,39 @@ class Table : private Noncopyable {
 
   std::vector<IndexStatistics> indexes_statistics() const;
 
+  /**
+   * Creates a subclass of AbstractTableIndex on a set of chunks of a specific column and adds the index to the
+   * table's index statistics.
+   * Table indexes can only be created on a set of immutable chunks.
+   */
+  template <typename Index>
+  void create_table_index(const ColumnID column_id, const std::vector<ChunkID>& chunk_ids,
+                          const std::string& name = "") {
+    static_assert(std::is_base_of<AbstractTableIndex, Index>::value,
+                  "'Index' template argument is not an AbstractTableIndex");
+
+    SegmentIndexType index_type = get_index_type_of<Index>();
+
+    std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>> chunks_to_index;
+    chunks_to_index.reserve(chunk_ids.size());
+    for (auto chunk_id : chunk_ids) {
+      auto chunk = get_chunk(chunk_id);
+      Assert(!chunk->is_mutable(), "Cannot index mutable chunk");
+      chunks_to_index.push_back(std::make_pair(chunk_id, chunk));
+    }
+    auto index = std::make_shared<Index>(chunks_to_index, column_id);
+    _table_indexes.emplace_back(index);
+
+    // Currently, IndexStatistics only hold information about the indexed column but not the indexed chunks
+    // for both, table-based and chunk-based indexes.
```

Comment:
The index statistics should be extended accordingly.

Reply:
We added TableIndexStatistics therefore.